### PR TITLE
AM2R: Some hydro station logic cleanup

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/games/am2r/json_data/Hydro Station.txt [hunk 1]
```diff
@@ -761,7 +761,19 @@ Extra - minimap_data: [{'x': 39, 'y': 23}, {'x': 39, 'y': 24}, {'x': 39, 'y': 25
...
   * Extra - instance_id: 111244
   > Horizontal Dock to Lower Hydro Station Hub
-      Enabled Septogg Helpers or Hijump Wall
+      Any of the following:
+          Enabled Septogg Helpers
+          Any of the following:
+              # Get up w/o Septogg
+              Hi-Jump Boots or Space Jump
+              # Spider Method
+              Movement (Advanced) and Can Use Spring Ball
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/json_data/Hydro Station.txt:770]: Assume this should be spider.

File: randovania/games/am2r/json_data/Hydro Station.txt [hunk 2]
```diff
@@ -1200,11 +1229,37 @@ Extra - minimap_data: [{'x': 19, 'y': 23}, {'x': 19, 'y': 24}, {'x': 20, 'y': 23
...
+              # Go Through Vines
+              Gravity Suit
+              Damage Boost (Beginner) and Lava Damage ≥ 60
   > Event - Alpha
       All of the following:
           Defeat dodging Alpha
-          Go Through Vines or Hijump With Power Grip Wall
+          Any of the following:
+              # Go Through Vines -  bottom way
+              Gravity Suit or Can Use Any Bombs
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/json_data/Hydro Station.txt:1243]: confused at why this is `can use any bombs`

File: randovania/games/am2r/json_data/Hydro Station.txt [hunk 3]
```diff
@@ -1200,11 +1229,37 @@ Extra - minimap_data: [{'x': 19, 'y': 23}, {'x': 19, 'y': 24}, {'x': 20, 'y': 23
...
+                      Can Use Spider Ball
+                      Mid-Air Morph (Intermediate) or Can Use Bombs or Can Use Spring Ball
+                  # WJ Method
+                  Morph Ball and Morph Glide (Intermediate) and Walljump (Advanced)
+                  # CBJ method
+                  Charged Bomb Jump (Advanced) and Can Use Charged Bomb Jump
+                  All of the following:
+                      # IBJ method
+                      Can Use Bombs
+                      Diagonal Infinite Bomb Jumping (Advanced) or Infinite Bomb Jumping (Advanced)
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/json_data/Hydro Station.txt:1262]: Assuming this `or` is intended as you can either go high enough then over with IBJ or direct with DIBJ.

Comment (Miepee) [reply randovania/games/am2r/json_data/Hydro Station.txt:1262]: correct.

**Review by JeffGainsNGames — APPROVED** [review]

**Pull request merged**